### PR TITLE
add UniverDocumentProcessor's output model

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

**Comment by vicancy** [comment]
:shipit:

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        [YamlMember(Alias = "seealso")]
+        [JsonProperty("seealso")]
+        public List<ApiLinkInfoBuildOutput> SeeAlsos { get; set; }
+
+        [YamlMember(Alias = "see")]
+        [JsonProperty("see")]
+        public List<ApiLinkInfoBuildOutput> Sees { get; set; }
+
+        [YamlMember(Alias = Constants.PropertyName.Inheritance)]
+        [MergeOption(MergeOption.Ignore)]
```
Comment (vwxyzh) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs]: remove

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+namespace Microsoft.DocAsCode.Build.UniversalReference
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Newtonsoft.Json;
+    using YamlDotNet.Serialization;
+
+    using Microsoft.DocAsCode.DataContracts.Common;
+    using Microsoft.DocAsCode.Common.EntityMergers;
```
Comment (vwxyzh) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs]: remove

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiInheritanceTreeBuildOutput.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    [Serializable]
+    public class ApiInheritanceTreeBuildOutput
+    {
+        [YamlMember(Alias = "type")]
+        [JsonProperty("type")]
+        public ApiNames Type { get; set; }
+
+        [YamlMember(Alias = Constants.PropertyName.Inheritance)]
+        [MergeOption(MergeOption.Ignore)]
```
Comment (vwxyzh) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiInheritanceTreeBuildOutput.cs]: remove

**Comment by vwxyzh** [comment]
:shipit:

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        [YamlMember(Alias = "summary")]
+        [JsonProperty("summary")]
+        public string Summary { get; set; } = null;
+
+        [YamlMember(Alias = "remarks")]
+        [JsonProperty("remarks")]
+        public string Remarks { get; set; }
+
+        [YamlMember(Alias = "example")]
+        [JsonProperty("example")]
```
Comment (hellosnow) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs]: >example [](start = 23, length = 7)

examples?

**Comment by hellosnow** [comment]
> namespace Microsoft.DocAsCode.Build.UniversalReference.BuildOutputs

copyright header

---
Refers to: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiInheritanceTreeBuildOutput.cs:1 in 1ae4a73. [](commit_id = 1ae4a73817205348fef9eb1bad075862caff7c5b, deletion_comment = False)

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.UniversalReference
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Newtonsoft.Json;
```
Comment (hellosnow) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs]: >Newtonsoft [](start = 10, length = 10)

minor: ns sort

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiInheritanceTreeBuildOutput.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    using System;
+    using System.Collections.Generic;
+
+    using Microsoft.DocAsCode.DataContracts.Common;
+
+    using Newtonsoft.Json;
+    using YamlDotNet.Serialization;
+
+    [Serializable]
+    public class ApiInheritanceTreeBuildOutput
```
Comment (hellosnow) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiInheritanceTreeBuildOutput.cs]: >ApiInheritanceTreeBuildOutput [](start = 17, length = 29)

will it have metadata?

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs (hunk 4 above)
Comment (superyyrrzz) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiBuildOutput.cs]: want to keep the same name with the one in mref

---
In reply to: [113360636](https://github.com/dotnet/docfx/pull/1596#discussion_r113360636) [](ancestors = 113360636)

File: plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiInheritanceTreeBuildOutput.cs (hunk 6 above)
Comment (superyyrrzz) [inline plugins/Microsoft.DocAsCode.Build.UniversalReference/BuildOutputs/ApiInheritanceTreeBuildOutput.cs]: yes, should have

---
In reply to: [113361121](https://github.com/dotnet/docfx/pull/1596#discussion_r113361121) [](ancestors = 113361121)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**